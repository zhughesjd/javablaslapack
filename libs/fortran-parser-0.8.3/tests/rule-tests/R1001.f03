! Test format-stmt
!      format-stmt  is  FORMAT format-specification
!   
!      format-specification  is  ( [ format-item-list ] )
!
! Not [completely] tested here: format-item-list.
000 format (I12, i1)
001 format (1pe12.4)
002 format (1PE12.4, I10)
003 format (I12 / 'Dates:', 22I3)
004 format (1x, ''isn''''t'', 1x)
005 format(//)
006 format(:)
007 format(:::)
008 format(::)
009 format(I12, /, ' Dates: ', 2 (2I3, I5))
010 format (I12 / 'Dates:', (2I3, I5))
011 format (I12 / 'Dates:', 2(2I3, I5))
012 format (1x, i1, 1x, 'isn''t', 1x, i1)
013 format(1x, 2(f10.3, i5))
014 format (1x, f10.3, i5, f10.3, i5, f10.3, i5)

end

