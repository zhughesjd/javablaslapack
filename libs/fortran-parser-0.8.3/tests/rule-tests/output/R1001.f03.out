*******************************************
args[1]: R1001.f03
R1101:main-program__begin:
R 313:label: lbl=[@12,186:188='000',<10>,7:0]
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='i12',<194>,7:12] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='i1',<194>,7:16] hasFormatItemList=false
R1003:format-item-list: count=2
R1002:format-specification: hasFormatItemList=true
R1001:format-stmt: label=[@12,186:188='000',<10>,7:0]
R 207:declaration-construct:
R 313:label: lbl=[@23,207:209='001',<10>,8:0]
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='1pe12.4',<195>,8:12] hasFormatItemList=false
R1003:format-item-list: count=1
R1002:format-specification: hasFormatItemList=true
R1001:format-stmt: label=[@23,207:209='001',<10>,8:0]
R 207:declaration-construct:
R 313:label: lbl=[@33,228:230='002',<10>,9:0]
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='1pe12.4',<195>,9:12] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='i10',<194>,9:20] hasFormatItemList=false
R1003:format-item-list: count=2
R1002:format-specification: hasFormatItemList=true
R1001:format-stmt: label=[@33,228:230='002',<10>,9:0]
R 207:declaration-construct:
R 313:label: lbl=[@46,254:256='003',<10>,10:0]
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='i12',<194>,10:12] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='/',<195>,10:15] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0=''dates:'',<196>,10:16] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='22i3',<194>,10:25] hasFormatItemList=false
R1003:format-item-list: count=4
R1002:format-specification: hasFormatItemList=true
R1001:format-stmt: label=[@46,254:256='003',<10>,10:0]
R 207:declaration-construct:
R 313:label: lbl=[@62,288:290='004',<10>,11:0]
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='1x',<195>,11:12] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='''isn''''t''',<196>,11:15] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='1x',<195>,11:28] hasFormatItemList=false
R1003:format-item-list: count=3
R1002:format-specification: hasFormatItemList=true
R1001:format-stmt: label=[@62,288:290='004',<10>,11:0]
R 207:declaration-construct:
R 313:label: lbl=[@82,322:324='005',<10>,12:0]
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='/',<195>,12:11] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='/',<195>,12:12] hasFormatItemList=false
R1003:format-item-list: count=2
R1002:format-specification: hasFormatItemList=true
R1001:format-stmt: label=[@82,322:324='005',<10>,12:0]
R 207:declaration-construct:
R 313:label: lbl=[@89,337:339='006',<10>,13:0]
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='',<195>,13:11] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0=':',<195>,13:11] hasFormatItemList=false
R1003:format-item-list: count=2
R1002:format-specification: hasFormatItemList=true
R1001:format-stmt: label=[@89,337:339='006',<10>,13:0]
R 207:declaration-construct:
R 313:label: lbl=[@96,351:353='007',<10>,14:0]
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='',<195>,14:11] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0=':',<195>,14:11] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='',<195>,14:12] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0=':',<195>,14:12] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='',<195>,14:13] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0=':',<195>,14:13] hasFormatItemList=false
R1003:format-item-list: count=6
R1002:format-specification: hasFormatItemList=true
R1001:format-stmt: label=[@96,351:353='007',<10>,14:0]
R 207:declaration-construct:
R 313:label: lbl=[@104,367:369='008',<10>,15:0]
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='',<195>,15:11] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0=':',<195>,15:11] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='',<195>,15:12] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0=':',<195>,15:12] hasFormatItemList=false
R1003:format-item-list: count=4
R1002:format-specification: hasFormatItemList=true
R1001:format-stmt: label=[@104,367:369='008',<10>,15:0]
R 207:declaration-construct:
R 313:label: lbl=[@111,382:384='009',<10>,16:0]
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='i12',<194>,16:11] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='/',<195>,16:15] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='' dates: '',<196>,16:17] hasFormatItemList=false
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='2i3',<194>,16:30] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='i5',<194>,16:34] hasFormatItemList=false
R1003:format-item-list: count=2
R1003:format-item: descOrDigit=[@-1,0:0='2',<10>,16:28] hasFormatItemList=false
R1003:format-item-list: count=4
R1002:format-specification: hasFormatItemList=true
R1001:format-stmt: label=[@111,382:384='009',<10>,16:0]
R 207:declaration-construct:
R 313:label: lbl=[@135,426:428='010',<10>,17:0]
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='i12',<194>,17:12] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='/',<195>,17:15] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0=''dates:'',<196>,17:16] hasFormatItemList=false
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='2i3',<194>,17:26] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='i5',<194>,17:30] hasFormatItemList=false
R1003:format-item-list: count=2
R1003:format-item: descOrDigit=null hasFormatItemList=false
R1003:format-item-list: count=4
R1002:format-specification: hasFormatItemList=true
R1001:format-stmt: label=[@135,426:428='010',<10>,17:0]
R 207:declaration-construct:
R 313:label: lbl=[@156,465:467='011',<10>,18:0]
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='i12',<194>,18:12] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='/',<195>,18:15] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0=''dates:'',<196>,18:16] hasFormatItemList=false
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='2i3',<194>,18:27] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='i5',<194>,18:31] hasFormatItemList=false
R1003:format-item-list: count=2
R1003:format-item: descOrDigit=[@-1,0:0='2',<10>,18:25] hasFormatItemList=false
R1003:format-item-list: count=4
R1002:format-specification: hasFormatItemList=true
R1001:format-stmt: label=[@156,465:467='011',<10>,18:0]
R 207:declaration-construct:
R 313:label: lbl=[@178,505:507='012',<10>,19:0]
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='1x',<195>,19:12] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='i1',<194>,19:15] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='1x',<195>,19:18] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0=''isn''t'',<196>,19:21] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='1x',<195>,19:30] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='i1',<194>,19:33] hasFormatItemList=false
R1003:format-item-list: count=6
R1002:format-specification: hasFormatItemList=true
R1001:format-stmt: label=[@178,505:507='012',<10>,19:0]
R 207:declaration-construct:
R 313:label: lbl=[@204,547:549='013',<10>,20:0]
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='1x',<195>,20:11] hasFormatItemList=false
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='f10.3',<194>,20:16] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='i5',<194>,20:22] hasFormatItemList=false
R1003:format-item-list: count=2
R1003:format-item: descOrDigit=[@-1,0:0='2',<10>,20:14] hasFormatItemList=false
R1003:format-item-list: count=2
R1002:format-specification: hasFormatItemList=true
R1001:format-stmt: label=[@204,547:549='013',<10>,20:0]
R 207:declaration-construct:
R 313:label: lbl=[@222,576:578='014',<10>,21:0]
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='1x',<195>,21:12] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='f10.3',<194>,21:15] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='i5',<194>,21:21] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='f10.3',<194>,21:24] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='i5',<194>,21:30] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='f10.3',<194>,21:33] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='i5',<194>,21:39] hasFormatItemList=false
R1003:format-item-list: count=7
R1002:format-specification: hasFormatItemList=true
R1001:format-stmt: label=[@222,576:578='014',<10>,21:0]
R 207:declaration-construct:
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=15
R1103:end-program-stmt: id=null
R1101:main-program: hasProgramStmt=false hasExecutionPart=false hasInternalSubprogramPart=false
Parser exiting normally
********************************************
